src/main/java/reposense/git/GitUtil.java

```diff
@@ -79,15 +89,51 @@ static String convertToGitFormatsArgs(List<Format> formats) {
 
     /**
      * Returns the {@code String} command to specify the globs to exclude for `git log` command.
+     * Also checks that every file in {@code ignoreGlobList} is within the given repository {@code root}
      */
-    static String convertToGitExcludeGlobArgs(List<String> ignoreGlobList) {
+    public static String convertToGitExcludeGlobArgs(File root, List<String> ignoreGlobList) {
         StringBuilder gitExcludeGlobArgsBuilder = new StringBuilder();
         final String cmdFormat = " " + addQuote(":(exclude)%s");
         ignoreGlobList.stream()
-                .filter(item -> !item.isEmpty())
+                .filter(item -> !item.isEmpty() && isValidPath(root, item))
                 .map(ignoreGlob -> String.format(cmdFormat, ignoreGlob))
                 .forEach(gitExcludeGlobArgsBuilder::append);
 
         return gitExcludeGlobArgsBuilder.toString();
     }
+
+    /**
+     * Returns true if the {@code path} is inside the current repository
+     * Produces log messages when the invalid file path is skipped
+     */
+    public static boolean isValidPath(File repoRoot, String path) {
```

Comment:
seems like this method is only used internally, can set to private in that case.